sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -146,36 +170,14 @@ public void describe(SensorDescriptor descriptor) {
       .name("XML Sensor");
   }
 
-  private Optional<RuleKey> getParsingErrorKey() {
-    return checks.all().stream()
-      .filter(ParsingErrorCheck.class::isInstance)
-      .map(ParsingErrorCheck.class::cast)
-      .map(checks::ruleKey)
-      .findFirst();
-  }
-
-  private static void processParseException(ParseException e, SensorContext context, InputFile inputFile, Optional<RuleKey> parsingErrorKey) {
+  private void processParseException(Exception e, SensorContext context, InputFile inputFile) {
     reportAnalysisError(e, context, inputFile);
 
-    LOG.warn("Unable to parse file {}", inputFile.uri());
-    LOG.warn("Cause: {}", e.getMessage());
-
-    if (parsingErrorKey.isPresent()) {
-      // the ParsingErrorCheck rule is activated: we create a beautiful issue
-      NewIssue newIssue = context.newIssue();
-      NewIssueLocation primaryLocation = newIssue.newLocation()
-        .message("Parse error: " + e.getMessage())
-        .on(inputFile);
-      newIssue
-        .forRule(parsingErrorKey.get())
-        .at(primaryLocation)
-        .save();
-    }
-  }
+    logError(String.format("Unable to analyse file %s", inputFile.uri()), e);
 
-  private static void processException(Exception e, SensorContext context, InputFile inputFile) {
-    reportAnalysisError(e, context, inputFile);
-    LOG.error("Unable to analyse file " + inputFile.uri(), e);
+    if (reportingParsingErrors) {
```

Comment:
should we rename it to `parsingErrorCheckEnabled`? because it's not crystal clear what do we mean by "reporting" here